Framework/Core/include/Framework/ConfigContext.h

```diff
@@ -24,12 +24,16 @@ namespace framework
 class ConfigContext
 {
  public:
-  ConfigContext(ConfigParamRegistry& options) : mOptions{options} {}
+  ConfigContext(ConfigParamRegistry& options, bool helpasked = false) : mOptions{options}, mIsHelpAsked{helpasked} {}
```

Comment:
Maybe there will be more use cases in the future, I would pass the command line arguments an leave the handling to the costomization function.
```suggestion
  ConfigContext(ConfigParamRegistry& options, int argc, const char** argv) : mOptions{options}, mArgv{argv}, mArgc{argc} {}
```

Reply:
We can do that in a next iteration, though.

Reply:
Excellent. This is actually my preferred way as well.